### PR TITLE
feat(app): add fixture list view on desktop

### PR DESCRIPTION
**Pull request by jerader**
closes RAUT-694

# Overview

This PR creates a fixture list view on the desktop app and changes up the modules a bit to align more with designs.

# Test Plan

Turn the deck config ff on and upload a protocol to the app for the Flex. Look at the module & deck view and see that the fixture list items match designs. The update deck button logic will be wired up in a separate ticket as well as the instructions modal

Turn off the ff and see that the fixture info is hidden

Screen Shot 2023-10-04 at 12 15 47 PM

Screen Shot 2023-10-05 at 11 35 47 AM

# Changelog

- add fixture logic to `ProtocolSetup`, `SetupModules` and create `SetupFixtureList`
- create a util to grab the fixture images
- add test coverage
- small design tweaks to the module setup list view
- rename the `SetupModules` folder and component to `SetupModulesAndDeck`

# Review requests

see test plan

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx [hunk 1]
```diff
@@ -74,43 +117,43 @@ export function ProtocolRunSetup({
...
+  let moduleDescription: string = t(`${MODULE_SETUP_KEY}_description`, {
+    count: modules.length,
+  })
+  if (!hasModules) {
+    moduleDescription = i18n.format(t('no_modules_specified'), 'capitalize')
+  } else if (isOT3 && enableDeckConfig && (hasModules || hasFixtures)) {
+    moduleDescription = t('install_modules_and_fixtures')
+  } else if (isOT3 && enableDeckConfig && !hasModules && !hasFixtures) {
+    moduleDescription = t('no_modules_or_fixtures')
+  }
```
Comment (jerader) [inline app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx:168]: logic here is a bit confusing with the feature flag, can definitely be refactored when the ff is removed.

File: app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx [hunk 2]
```diff
@@ -210,7 +263,9 @@ export function ProtocolRunSetup({
             stepsKeysInOrder.map((stepKey, index) => (
               <Flex flexDirection={DIRECTION_COLUMN} key={stepKey}>
                 {(stepKey === 'liquid_setup_step' && !hasLiquids) ||
-                (stepKey === 'module_setup_step' && !hasModules) ? (
+                (stepKey === 'module_setup_step' &&
+                  (!hasModules ||
+                    (enableDeckConfig && !hasModules && !hasFixtures))) ? (
                   <EmptySetupStep
                     title={t(`${stepKey}_title`)}
```
Comment (brenthagen) [inline app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx]: same as below for this title, maybe worth pulling this title out as a const

                    isOT3 && stepKey === MODULE_SETUP_KEY && enableDeckConfig
                        ? `module_and_deck_setup`
                        : `${stepKey}_title`

File: app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx [hunk 3]
```diff
@@ -210,7 +263,9 @@ export function ProtocolRunSetup({
             stepsKeysInOrder.map((stepKey, index) => (
               <Flex flexDirection={DIRECTION_COLUMN} key={stepKey}>
                 {(stepKey === 'liquid_setup_step' && !hasLiquids) ||
-                (stepKey === 'module_setup_step' && !hasModules) ? (
+                (stepKey === 'module_setup_step' &&
+                  (!hasModules ||
+                    (enableDeckConfig && !hasModules && !hasFixtures))) ? (
```
Comment (brenthagen) [inline app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx]: always true when `!hasModules`

need to add `!enableDeckConfig` to the first: 
```
                  ((!enableDeckConfig && !hasModules) ||
                    (enableDeckConfig && !hasModules && !hasFixtures))) ? (
```

File: app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx [hunk 4]
```diff
@@ -111,6 +153,19 @@ export function ProtocolRunSetup({
   if (robot == null) return null
   const hasLiquids = protocolData != null && protocolData.liquids?.length > 0
   const hasModules = protocolData != null && modules.length > 0
+  const hasFixtures =
+    protocolData != null && Object.keys(STUBBED_LOAD_FIXTURE_BY_SLOT).length > 0
+
+  let moduleDescription: string = t(`${MODULE_SETUP_KEY}_description`, {
+    count: modules.length,
+  })
+  if (!hasModules) {
```
Comment (brenthagen) [inline app/src/organisms/Devices/ProtocolRun/ProtocolRunSetup.tsx]: also need a `!enableDeckConfig` here

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/__tests__/SetupModules.test.tsx [hunk 5]
```diff
@@ -1,20 +1,42 @@
 import * as React from 'react'
```
Comment (brenthagen) [inline app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/__tests__/SetupModules.test.tsx]: file name: SetupModuleAndDeck.test.tsx

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/SetupFixtureList.tsx [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+  cutout,
+  commandId,
+}: FixtureListItemProps): JSX.Element {
+  const { t } = useTranslation('protocol_setup')
+  const configuration = useLoadedFixturesConfigStatus(loadedFixtures)
+  const configurationStatus = configuration.find(
+    config => config.id === commandId
+  )?.configurationStatus
+
+  let statusLabel
```
Comment (brenthagen) [inline app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/SetupFixtureList.tsx:117]: it's great to see the status label wired up to the configurator!

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/SetupFixtureList.tsx [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+  CONFIGURED,
+} from '../../../../resources/deck_configuration/hooks'
+import { StyledText } from '../../../../atoms/text'
+import { StatusLabel } from '../../../../atoms/StatusLabel'
+import { TertiaryButton } from '../../../../atoms/buttons/TertiaryButton'
+import { getFixtureImage } from './utils'
+
+import type { LoadedFixturesBySlot } from '@opentrons/api-client'
+
+interface SetupModulesListProps {
```
Comment (brenthagen) [inline app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/SetupFixtureList.tsx]: `SetupFixtureListProps`

**Review by brenthagen — COMMENTED** [review]
like on ODD, we'll need to show fixture location conflicts with modules in SetupModulesList.tsx (takes precedence over calibration and connection status). but could leave this for the chunk of work in https://opentrons.atlassian.net/browse/RAUT-700

File: app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/__tests__/SetupModules.test.tsx (hunk 5 above)
Comment (jerader) [reply app/src/organisms/Devices/ProtocolRun/SetupModuleAndDeck/__tests__/SetupModules.test.tsx]: oops good catch!

**Comment by jerader** [comment]
> like on ODD, we'll need to show fixture location conflicts with modules in SetupModulesList.tsx (takes precedence over calibration and connection status). but could leave this for the chunk of work in https://opentrons.atlassian.net/browse/RAUT-700

@brenthagen, oh ya, I wasn't sure if that was part of this ticket. I'm down to add it as part of RAUT-700

**Review by brenthagen — APPROVED** [review]

**Pull request merged**